devops/setup-gpg/action.yaml

```diff
@@ -0,0 +1,48 @@
+name: 'Setup GPG'
+description: 'Configures this action to run gpg with a given key and pass'
+inputs:
+  private-key:  # id of input
+    description: 'gpg private key'
+    required: true
+  key-pass:  # id of input
+    description: 'gpg key pass'
+    required: true
+  key-name:  # id of input
+    description: 'gpg key name'
+    required: true
+    default: "Aerospike"
+runs:
+  using: "composite"
+  steps:
+      - name: Set up GPG
+        env:
+          GPG_PRIVATE_KEY: ${{ inputs.private-key }}
+          GPG_KEY_PASS: ${{ inputs.key-pass }}
+          GPG_ID: ${{ inputs.key-name }}
+        run: |
+          # Setup gpg
+          mkdir -p ~/.gnupg
+          echo "$GPG_PRIVATE_KEY" | gpg --import --batch --yes
+          echo "$GPG_KEY_PASS"
+
+          # configure for non-interactive use
+          export GPG_TTY=no-tty
+          echo -e "pinentry-mode loopback\nuse-agent" >> ~/.gnupg/gpg.conf
+          echo -e "allow-loopback-pinentry" >> ~/.gnupg/gpg-agent.conf
+
+          # configure rpm's
+          echo -e "$GPG_KEY_PASS" >> ~/pass
+          echo -e "%_signature gpg" >> ~/.rpmmacros
+          echo -e "%_gpg_path ~/.gnupg" >> ~/.rpmmacros
+          echo -e "%_gpg_name $GPG_ID" >> ~/.rpmmacros
+          echo -e "%_gpgbin /usr/bin/gpg" >> ~/.rpmmacros
```

Comment:
How do you know for sure that `/usr/bin/gpg` is where `gpg` is always installed? Maybe `$(which gpg)` is safer?

Reply:
so I don't know for this one to be honest. maybe you are right but also I don't want to use a different one on accident? right? I'm going to leave this for now but you might be right.